src/components/NcBlurHash/NcBlurHash.vue

```diff
@@ -0,0 +1,260 @@
+				:hash="blurHash"
+				:src="imageSource"
+				@load="onLoaded" />
+
+			<NcButton @click="toggleImage">
+				{{
+					loading
+						? 'Loading...'
+						: (loaded ? 'Unload image' : 'Load image')
+				}}
+			</NcButton>
+		</div>
+	</template>
+	<script>
+	export default {
+		data() {
+			return {
+				loaded: false,
+				loading: false,
+				blurHash: 'M8CR]OkDD%kD9ZtRayofaykC00ay$_ay~T',
+			}
+		},
+		computed: {
+			// This is cheating but we can not emulate slow network connection
+			// so imagine that this means the source becomes loaded
+			imageSource() {
+				return this.loaded
+					? 'favicon-touch.png'
+					: 'invalid-file-that-will-never-load.png'
+			},
+		},
+		methods: {
+			toggleImage() {
+				if (this.loaded) {
+					this.loaded = false
+					this.loading = false
+				} else {
+					// emulate slow network
+					this.loading = true
+					window.setTimeout(() => {
+						this.loaded = !this.loaded
+						this.loading = false
+					}, 3000)
+				}
+			},
+
+			// you could use `success` here (boolean) to decide if the image is loaded or failed
+			onLoaded(success) {
+				// ...
+			},
+		},
+	}
+	</script>
+	<style scoped>
+		.wrapper {
+			display: flex;
+			flex-direction: row;
+			align-items: center;
+			gap: 12px;
+		}
+		.shown-image {
+			width: 150px;
+			height: 150px;
+			border-radius: 24px;
+		}
+	</style>
+```
+
+### Manual usage as a placeholder
+
+Using `v-if` is also possible, this can e.g. used if the image is not loaded from an URL.
+
+```vue
+	<template>
+		<div class="wrapper">
+			<img :class="loaded ? 'shown-image' : 'hidden-visually'"
+				alt=""
+				src="favicon-touch.png">
+			<NcBlurHash v-if="!loaded"
+				class="shown-image"
+				:hash="blurHash" />
+			<NcButton @click="loaded = !loaded">Toggle blur hash</NcButton>
+		</div>
+	</template>
+	<script>
+	export default {
+		data() {
+			return {
+				loaded: false,
+				blurHash: 'M8CR]OkDD%kD9ZtRayofaykC00ay$_ay~T',
+			}
+		},
+	}
+	</script>
+	<style scoped>
+		.wrapper {
+			display: flex;
+			flex-direction: row;
+			align-items: center;
+			gap: 12px;
+		}
+		.shown-image {
+			width: 150px;
+			height: 150px;
+			border-radius: 24px;
+		}
+	</style>
+```
+</docs>
+
+<script setup>
+import { decode } from 'blurhash'
+import { ref, watch, nextTick } from 'vue'
+import { logger } from '../../utils/logger.ts'
+import { preloadImage } from '../../functions/preloadImage/index.ts'
+
+const props = defineProps({
+	/**
+	 * The blur hash value to use.
+	 */
+	hash: {
+		required: true,
+		type: String,
+	},
+
+	/**
+	 * This is normally not needed, but if this blur hash is not only intended
+	 * for decorative purpose, descriptive text should be passed for accessibility.
+	 */
+	alt: {
+		type: String,
+		default: '',
+	},
+
+	/**
+	 * Optional an image source to load, during the load the blur hash is shown.
+	 * As soon as it is loaded the image will be shown instead.
+	 */
+	src: {
+		type: String,
+		default: '',
+	},
```

Comment:
Shouldn't this be handled by the parent component? The usage in Files and Talk is quite straightforward, but in Photos, we have a more involved rendering logic and this won't work.

Reply:
But this could remove the need for that logic in Files and Talk.
Photos can still have their own logic using the "manual" step as show in the "manual usage" example